src/test/java/test/wsdl/soap12/assertion/Soap12TestDocBindingImpl.java

```
@@ -9,6 +9,78 @@
+                                if (isSupported) {
+                                    inputHeader = headerElement;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if (inputHeader != null) {
+                    // Try different ways to get the header value
+                    String headerValue = null;
+                    try {
+                        Object objValue = inputHeader.getObjectValue();
+                        if (objValue != null) {
+                            headerValue = objValue.toString();
+                        } else {
+                            // Try getValue() method instead
+                            headerValue = inputHeader.getValue();
+                        }
+                    } catch (Exception e) {
+                        headerValue = inputHeader.getValue();
+                    }
+
+
+                    if (headerValue != null) {
+                        // Add response header "responseOk" with the same value and role
+                        org.apache.axis.message.SOAPEnvelope respEnv = mc.getResponseMessage().getSOAPEnvelope();
+                        org.apache.axis.message.SOAPHeaderElement responseHeader =
+                            new org.apache.axis.message.SOAPHeaderElement("http://example.org/ts-tests", "responseOk", headerValue);
+
+                        // Copy the role from the input header to maintain SOAP 1.2 compatibility
+                        String inputRole = inputHeader.getRole();
+                        if (inputRole != null) {
+                            responseHeader.setRole(inputRole);
+                        }
+
+                        respEnv.addHeader(responseHeader);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            // Silently handle header processing errors
```

This catch feels very wrong. Why even bother trying to read the headers if we're going to silently swallow all errors?

---

Updated it to throw proper exception.